### ISSUE TITLE
Changing final variable no warning

### ISSUE THREAD

**Issue by GF-Huang** [microsoft/pylance-release#6447]
## Environment data

-   Pylance version: `v2024.9.2`
-   OS and version: `Win11 23H2`
-   Python version (& distribution if applicable, e.g. Anaconda): `Conda Python 3.11.9`

## Code Snippet

```python
from typing import Final

test: Final[int] = 0
test = 1
```

![image](https://github.com/user-attachments/assets/c5c99b7a-5b36-4834-a462-f7451f2eac7b)

## Repro Steps

1. XXX

## Expected behavior

Show me some warning like: "you can't change constant".

## Actual behavior

No warning.

## Logs

```
XXX
```

**Comment by rchiodo** [comment]
Thanks for the issue. You have to set your typechecking mode higher than 'off' for an error to show up:


This is what I get with `typeCheckingMode` set to `basic`
![Image](https://github.com/user-attachments/assets/8e8d5cec-edfd-4432-b983-ff2ab7b11129)

**Comment by GF-Huang** [comment]
> Thanks for the issue. You have to set your typechecking mode higher than 'off' for an error to show up:
> 
> This is what I get with `typeCheckingMode` set to `basic`

Is it not works for jupyter notebook? I had set to `basic` but not work, same as previous.

**Comment by rchiodo** [comment]
It works for me in a jupyter notebook. Is this an untitled notebook? The setting applies to things in the workspace, it might be the untitled notebook is not considered part of the same settings.

**Comment by rchiodo** [comment]
It does work for me in the untitled case too though. Must be the typecheckingmode is not being applied for some reason to your notebook.

**Comment by GF-Huang** [comment]
Not work. I had tried both untitled or saved file.

**Comment by rchiodo** [comment]
Can you include your logs after running 'Pylance : Start Logging'. It should tell us which workspace we think the notebook is in.

**Comment by GF-Huang** [comment]
Hope these helpful.

[pylance_2024.9.2_id_3.txt](https://github.com/user-attachments/files/17136578/pylance_2024.9.2_id_3.txt)


https://github.com/user-attachments/assets/64ccfa08-1a1a-4487-b99f-067c13b1d0d5

**Comment by rchiodo** [comment]
Hmm, I can reproduce it now too using workspace settings.json. Not sure why that matters.

**Comment by rchiodo** [comment]
Yeah there's an error reading the settings for some reason. Here:

```
2024-09-25 11:38:05.018 [info] [Error - 11:38:05 AM] (17092) Error reading settings: TypeError: Cannot set properties of undefined (setting 'strictListInference')
2024-09-25 11:38:05.019 [info] (17092) Notebook settings returned for workspace: file:///c%3A/Users/rchiodo/source/testing/notebooks: {
  "autoSearchPaths": false,
  "openFilesOnly": true,
  "watchForConfigChanges": false,
  "watchForSourceChanges": false,
  "watchForLibraryChanges": false,
  "typeCheckingMode": "off", <-- This is coming out wrong
  "diagnosticSeverityOverrides": {
    "reportMissingImports": "none",
    "reportMissingModuleSource": "none"
  },
  "diagnosticBooleanOverrides": {
    "enableReachabilityAnalysis": false
  },
  "enableExtractCodeAction": false,
  "callArgumentNameInlayHints": "off",
  "variableInlayTypeHints": false,
  "pytestParametersInlayTypeHints": false,
  "functionReturnInlayTypeHints": false,
  "aiCodeActions": {}
}
```

**Comment by rchiodo** [comment]
Figured out the problem internally. We're not propagating the imports from the first cell to the second cell.

If you put the import in the same cell, the error shows up:

```
from typing import Final

x: Final[int] = 3
x = 212
```

**Comment by rchiodo** [comment]
I can reproduce this with two python files (no need for a notebook). Should be a pyright issue? 

Create two files:

exporting_file.py:
```python
from typing import Final
```
importing_file.py
```python
from exporting_file import *

x: Final[int] = 1
x = 2 # should be an error here and there isn't
```
